sycl/include/sycl/reduction.hpp

```diff
@@ -1629,12 +1627,24 @@ void reduceReduLocalAccs(size_t IndexA, size_t IndexB,
   (ProcessOne(std::get<Is>(LocalAccs), std::get<Is>(BOPs)), ...);
 }
 
+template <typename... LocalAccT, typename... BOPsT, size_t... Is,
+          typename BarrierTy>
+void doTreeReduction(size_t WorkSize, size_t LID,
+                     ReduTupleT<LocalAccT...> &LocalAccs,
+                     ReduTupleT<BOPsT...> &BOPs,
+                     std::index_sequence<Is...> ReduIndices,
+                     BarrierTy Barrier) {
+  doTreeReductionHelper(WorkSize, LID, Barrier, [&](size_t I, size_t J) {
+    reduceReduLocalAccs(I, J, LocalAccs, BOPs, ReduIndices);
+  });
```

Comment:
Do we have the same memory consumption peak here? In other words, do we process WG Size dimension and the reductions list in the same order now as before?

Reply:
There should be no change to the work distribution. There may be a small change in operation order inside the tree reduction as there are no longer any results being stored in the additional element (which is no more.)